packages/expo-modules-core/ios/JSI/EXJSIConversions.mm

```diff
@@ -73,6 +84,8 @@
       return convertNSNumberToJSIBoolean(runtime, (NSNumber *)value);
     }
     return convertNSNumberToJSINumber(runtime, (NSNumber *)value);
+  } else if ([value isKindOfClass:[NSData class]]) {
```

Comment:
`NSData` is less common than `NSDictionary` or `NSArray`, so I would place it after them (small optimizations 🤷‍♂️)